src/segment_reshape/map_tool/segment_reshape_tool.py

```diff
@@ -58,82 +78,171 @@ def _optional_identify_tool(
             tool.deleteLater()
 
 
+COLOR_BLUE = QColor(10, 20, 150)
+COLOR_GREY = QColor(211, 211, 211)
+
+
 class ToolMode(Enum):
     PICK_SEGMENT = "pick_segment"
     RESHAPE = "reshape"
 
 
-class SegmentReshapeTool(QgsMapToolEdit):
+class AddVertexReturn(IntEnum):
+    Success = 0
+    TransformationError = 2
+
+
+class SegmentReshapeToolHandler(QgsAbstractMapToolHandler):
+    def __init__(self, tool: "SegmentReshapeTool", action: "QAction") -> None:
+        super().__init__(tool, action)
+
+    def isCompatibleWithLayer(  # noqa: N802
+        self, layer: "QgsMapLayer", context: QgsAbstractMapToolHandler.Context
+    ) -> bool:
+        return (
+            isinstance(layer, QgsVectorLayer)
+            and layer.geometryType()
+            in (
+                QgsWkbTypes.GeometryType.PolygonGeometry,
+                QgsWkbTypes.GeometryType.LineGeometry,
+            )
+            # and layer.isEditable()
```

Comment:
Is this line commented out by accident?

Reply:
That row is left there by accident. That is what I tested at first but that would change the current behavior. Currently the plugin is able to edit layers even if they are not in edit mode. That is however something I'm not sure we should allow.